backend/twitch-api/resource/stream-tags.js

```diff
@@ -14,108 +12,55 @@ const { snakeKeys } = require('js-convert-case');
+        const response = await client.helix.tags.getAllStreamTagsPaginated().getAll();
 
-        if (response == null || response.data == null || response.data.length < 1) {
-            return null;
+        if (response) {
+            /**@type {TwitchStreamTag[]} */
+            return response.filter(tag => !tag.isAuto).map(tag => mapTwitchTag(tag));
         }
-
-        return response;
     } catch (error) {
         logger.error("Failed to get all stream tags", error);
         return null;
     }
-}
-
-async function getAllStreamTagsPaginated() {
-    let response = await getAllStreamTags();
-    let cursor = "";
-    let streamTags = response.data;
-
-    while (response.pagination.cursor && response.pagination.cursor !== cursor) {
-        cursor = response.pagination.cursor;
-        response = await getAllStreamTags(cursor);
-        if (response == null) break;
-
-        streamTags = streamTags.concat(response.data.filter(tag => !tag.is_auto));
-    }
-
-    /**@type {TwitchStreamTag[]} */
-    return streamTags.map(tag => mapTwitchTag(tag));
-}
-
-async function getChannelStreamTags() {
-    const client = twitchApi.getClient();
+};
 
+const getChannelStreamTags = async () => {
     try {
-        const response = await client.callApi({
-            type: TwitchAPICallType.Helix,
-            url: "streams/tags",
-            method: "GET",
-            query: {
-                "broadcaster_id": accountAccess.getAccounts().streamer.userId
-            }
-        });
+        const client = twitchApi.getClient();
+        const response = await client.helix.streams.getStreamTags(accountAccess.getAccounts().streamer.userId);
 
-        if (response == null || response.data == null || response.data.length < 1) {
-            return null;
+        if (response) {
+            /**@type {TwitchStreamTag[]} */
+            return response.filter(tag => !tag.isAuto).map(tag => mapTwitchTag(tag));
         }
-
-        /**@type {TwitchStreamTag[]} */
-        return response.data.filter(tag => !tag.is_auto).map(tag => mapTwitchTag(tag));
     } catch (error) {
         logger.error("Failed to get channel stream tags", error);
         return null;
     }
-}
+};
 
-async function updateChannelStreamTags(tagIds) {
-    const client = twitchApi.getClient();
+const updateChannelStreamTags = async (tagIds) => {
     try {
-        await client.callApi({
-            type: TwitchAPICallType.Helix,
-            url: "streams/tags",
-            method: "PUT",
-            query: {
-                "broadcaster_id": accountAccess.getAccounts().streamer.userId
-            },
-            body: snakeKeys(tagIds)
-        });
-        return true;
+        const client = twitchApi.getClient();
+        await client.helix.streams.replaceStreamTags(accountAccess.getAccounts().streamer.userId, tagIds);
     } catch (error) {
         logger.error("Failed to update channel stream tags", error);
         return false;
```

Comment:
Returns false in the catch but the `return true` was removed from the try. Is the return boolean even used? If not, we can just remove the `return false` from the catch. If yes, we'll want to add the `return true` back to try.

Reply:
Fixed.